M2/Macaulay2/packages/AbstractSimplicialComplexes.m2

```diff
@@ -329,117 +288,172 @@ simplicialMakeMatrix(List,List) := (kPlusOneFaces,kFaces) -> (
 
 inducedReducedSimplicialChainComplexMap(AbstractSimplicialComplex,AbstractSimplicialComplex) := (L,H) -> (
     h := reducedSimplicialChainComplex H;
     l := reducedSimplicialChainComplex L;
-    if ((abstractSimplicialComplex {{}}) == H) then return map(l,h, hashTable {-2 => map(l_(-2),h_(-2),zero), -1 => map(l_(-1),h_(-1),id_(h_(-1)))})
-    else( 
-    f := hashTable apply(spots h, i -> if i == -1 then i => map(l_(-1),h_(-1),id_(h_(-1))) else i => inducedKFaceSimplicialChainComplexMap(i,L,H));
-    return map(l,h,f);
-    )
-)
-
------
- 
+    if ((abstractSimplicialComplex {{}}) == H)
+    then return map(l,h, hashTable {-2 => map(l_(-2),h_(-2),zero), -1 => map(l_(-1),h_(-1),id_(h_(-1)))})
+    else(myKeys := apply(spots h, i ->
+	    if i == -1 then i => map(l_(-1),h_(-1),id_(h_(-1)))
+	    else i => inducedKFaceSimplicialChainComplexMap(i,L,H));
+            f := hashTable myKeys;
+            return map(l,h,f);))
+--------------------------------------------------------------------------------------------------------------
+-------------------------------------------------------------------------------------------------------------- 
 -* Documentation section *-
 beginDocumentation()
 
-document { 
-  Key => AbstractSimplicialComplexes,
-  Headline => "a package for working with abstract simplicial complexes",
-  "In this package our conventions are that `abstract simplicial complexes' have vertices supported on the set [n] := {1,...,n}.
-  Our aim is to provide a methology for working with such objects directly.  
-  In this regard our approach differs from that of the package 
-  SimplicialComplexes.m2.
-  Here, we are especially interested in homological aspects of SimplicialComplexes and our approach is to implement such simplicial complexes as certain graded lists. 
-  In particular, we provide methods for working with the chain complexes that are associated to each abstract simplicial complex.  
-  We also give some functionality for producing random simplicial complexes.",
-
-   SUBSECTION "An overview of this package",
-   UL {
-    TO "How to make abstract simplicial complexes", 
-    TO "How to make reduced and non-reduced simplicial chain complexes",
-    TO "Calculations with random simplicial complexes"
-    },
-}
+------------------------------
+-- Documentation Index page --
+------------------------------
+
+doc ///  
+    Key
+        AbstractSimplicialComplexes
+    Headline 
+        a package for working with abstract simplicial complexes
+    Description
+        Text
+	    The package AbstractSimplicialComplexes provides a methodology for 
+            working with abstract simplicial complexes.  The starting point is to represent each
+	    abstract simplicial complex as a graded list.  In this regard our approach differs
+	    fundamentally from that of the package
+	    @HREF("https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/SimplicialComplexes/html/index.html","SimplicialComplexes.m2")@.
```

Comment:
You should link to another package with `@TO "SimplicialComplexes::SimplicialComplexes"@`.